flycheck-vale.el

```diff
@@ -132,7 +132,8 @@ passing the results to CALLBACK."
                               flycheck-vale-output-buffer
                               flycheck-vale-program
                               "--output"
-                              "JSON")))
+                              "JSON"
+                              buffer-file-name)))
```

Comment:
Changing this to the below will account for buffers that do not have an associated file.

If the buffer name doesn't have an extension it returns `""` which vale accepts and ignores any ML specific rules (such as for Org)

```
"--ext"
(file-name-extension (buffer-name) t)
```